WalletWasabi.Gui/ViewModels/StatusBarViewModel.cs

```diff
@@ -384,31 +402,31 @@ private void SetCustomStatusOrReady()
 		{
 			if (StatusQueue.Any())
 			{
-				Status = StatusQueue.Last();
+				StatusBacking = StatusQueue.Last();
 			}
 			else
 			{
-				Status = StatusBarStatus.Ready;
+				StatusBacking = StatusBarStatus.Ready;
 			}
 		}
 
 		private bool TrySetPriorityStatus()
 		{
```

Comment:
I understand how your approach fixes the threading issue. However it seems a bit of a long way round (perhaps there is a reason to do it this way?)

My suggestion is it might be much simpler to not have `StatusBacking` and post all Writes to Status to UIThread?

Reply:
Calling something asynchronously from inside a lock is kinda smells to me.

Reply:
well my assumption is you dont want you code from backend to block whilst it waits for the ui to update?

In any case I think your solution works well and im actually starting to like that pattern.

I wonder though it might then be nicer to get rid of Status property, and have an UpdateStatus method that you call from locked code? would fit better with your pattern, and not break the typical viewmodel pattern, where all the properties should really be ui compatible.

Reply:
That's the exact same suggestion, just encapsulating it into a function. It'd still call asynchronously on another thread from inside a lock.

Reply:
I removed the statusbacking property.